modules/text2vec-cohere/vectorizer/objects.go

```diff
@@ -14,31 +14,61 @@ package vectorizer
 import (
 	"context"
 	"fmt"
+	"time"
 
+	"github.com/sirupsen/logrus"
 	"github.com/weaviate/weaviate/entities/models"
+	"github.com/weaviate/weaviate/entities/modulecapabilities"
 	"github.com/weaviate/weaviate/entities/moduletools"
 	"github.com/weaviate/weaviate/modules/text2vec-cohere/ent"
 	objectsvectorizer "github.com/weaviate/weaviate/usecases/modulecomponents/vectorizer"
 	libvectorizer "github.com/weaviate/weaviate/usecases/vectorizer"
 )
 
+const (
+	MaxObjectsPerBatch = 96 // https://docs.cohere.com/reference/embed
+	MaxTimePerBatch    = float64(10)
+)
+
 type Vectorizer struct {
 	client           Client
 	objectVectorizer *objectsvectorizer.ObjectVectorizer
+	batchVectorizer  *modulecapabilities.Batch
 }
 
-func New(client Client) *Vectorizer {
+func New(client Client, logger logrus.FieldLogger) *Vectorizer {
+	// cohere has only request limit and no token limit
+	execAfterRequestFunction := func(limits *modulecapabilities.RateLimits) {
+		// refresh is after 60 seconds but leave a bit of room for errors. Otherwise we only deduct the request that just happened
+		if limits.LastOverwrite.Add(61 * time.Second).After(time.Now()) {
+			limits.RemainingRequests -= 1
+			return
+		}
+
+		// initial values
+		limits.RemainingRequests = 10000
+		limits.ResetTokens = 61
+		limits.LimitRequests = 10000
+		limits.LastOverwrite = time.Now()
+
+		// high dummy values
+		limits.RemainingTokens = 10000000
+		limits.LimitTokens = 10000000
+		limits.ResetTokens = 1
```

Comment:
those limit values, where did you get them?

Reply:
from cohere, added a comment with the url